integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -138,27 +162,47 @@ def from_dict(cls, data) -> "SagemakerGenerator":
         """
         Deserialize the dictionary into an instance of SagemakerGenerator.
         """
+        deserialize_secrets_inplace(
+            data["init_parameters"],
+            ["aws_access_key_id", "aws_secret_access_key", "aws_session_token", "aws_region_name", "aws_profile_name"],
+        )
         return default_from_dict(cls, data)
 
-    def warm_up(self):
+    @classmethod
+    def get_aws_session(
```

Comment:
I would change this to `staticmethod` since we don't really need the class in this case.
Also let's rename it to `_get_aws_session`, this should be part of the private API.